extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSConfig.java

```diff
@@ -6,69 +6,58 @@
 
 import io.quarkus.runtime.annotations.ConfigGroup;
 import io.quarkus.runtime.annotations.ConfigItem;
-import io.vertx.core.http.HttpMethod;
 
 @ConfigGroup
 public class CORSConfig {
 
     /**
-     * Origins allowed for CORS
-     *
-     * Comma separated list of valid URLs. ex: http://www.quarkus.io,http://localhost:3000
-     * The filter allows any origin if this is not set.
-     *
-     * default: returns any requested origin as valid
+     * Origins allowed for CORS.
+     * <p>
+     * Comma separated list of valid URLs e.g. http://www.quarkus.io,http://localhost:3000.
+     * <p>
+     * The filter allows any origin if not set.
      */
-    @ConfigItem
+    @ConfigItem(defaultValue = "*")
     public Optional<List<String>> origins;
 
     /**
-     * HTTP methods allowed for CORS
-     *
-     * Comma separated list of valid methods. ex: GET,PUT,POST
-     * The filter allows any method if this is not set.
-     *
-     * default: returns any requested method as valid
+     * HTTP methods allowed for CORS.
+     * <p>
+     * Comma separated list of valid methods e.g. GET,PUT,POST.
+     * <p>
+     * The filter allows any method if not set.
      */
-    @ConfigItem
-    public Optional<List<HttpMethod>> methods;
+    @ConfigItem(defaultValue = "*")
+    public Optional<List<String>> methods;
 
     /**
-     * HTTP headers allowed for CORS
-     *
-     * Comma separated list of valid headers. ex: X-Custom,Content-Disposition
-     * The filter allows any header if this is not set.
-     *
-     * default: returns any requested header as valid
+     * HTTP headers allowed for CORS.
+     * <p>
+     * Comma separated list of valid headers e.g. X-Custom,Content-Disposition.
+     * <p>
+     * The filter allows any header if not set.
      */
-    @ConfigItem
+    @ConfigItem(defaultValue = "*")
     public Optional<List<String>> headers;
 
     /**
-     * HTTP headers exposed in CORS
-     *
-     * Comma separated list of valid headers. ex: X-Custom,Content-Disposition
-     *
-     * default: empty
+     * HTTP headers exposed in CORS.
+     * <p>
+     * Comma separated list of exposed headers e.g. X-Custom,Content-Disposition.
      */
-    @ConfigItem
+    @ConfigItem(defaultValue = "*")
     public Optional<List<String>> exposedHeaders;
```

Comment:
@Postremus sorry I'm getting back to this one a bit late. I rebased and added a commit to improve the documentation formatting.

And while doing so, I saw this change. It looks like a significant change in behavior or am I wrong? Did it really expose all headers by default before?

Reply:
@Postremus ping?

Reply:
@gsmet pong :)
You are right. That is quite the change from previous behavior. I removed this change.